internal/util/id.go

```diff
@@ -0,0 +1,25 @@
+package util
+
+import (
+	"fmt"
+	"strconv"
+)
+
+func CastInt64(v any) int64 {
+	switch o := v.(type) {
+	case int64:
+		return o
+	case int:
+		return int64(o)
+	default:
+		panic(fmt.Sprintf("unexpected type %T for %#v", v, v))
+	}
+}
+
+func FormatID[T ~int | ~int32 | ~int64](v T) string {
```

Comment:
Is `~int32` actually used anywhere? This would always result in a runtime panic in `CastInt64()`, so maybe returning a compile-time error instead would be easier to spot?